### PR TITLE
Add macOS Sequoia local networks services requirements

### PR DESCRIPTION
**Pull request by Pythyu**
### What does this PR do?

Add a message inside of dmg `info.plist` file to justify why our agent needs access to local networks

### Motivation

Following latest macOS Sequoia update, our agent installed in user-mode is asking for local network access via a pop-up. Current agent versions needs to be installed in system-wide to avoid the pop-up. As an additional mitigation add a reason why we need local network access
Full [details](https://docs.google.com/document/d/10vAfd-W61dogbVyBl_cM3wSYYdODBimU8WEpE04iSSM/edit?tab=t.0#heading=h.ixldy9ty2xy5)

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: omnibus/config/templates/datadog-agent-mac-app/Info.plist.erb [hunk 1]
```diff
@@ -32,5 +32,7 @@
     <string>Copyright (c) 2009-<%= year %>, Datadog Inc.</string>
     <key>NSPrincipalClass</key>
     <string>NSApplication</string>
+    <key>NSLocalNetworkUsageDescription</key>
+    <string>Datadog need access to local network services to function"</string>
```
Comment (FlorentClarret) [inline omnibus/config/templates/datadog-agent-mac-app/Info.plist.erb:36]: **🥜 nitpick:** ‏

```suggestion
    <string>Datadog needs access to local network services to function"</string>
```

File: releasenotes/notes/macos-sequoia-pop-up-fix-ffbb1b2447b059cd.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Agent running as user on macOS Sequoia will no longer send a pop-up notification asking for local network access.
```
Comment (drichards-87) [inline releasenotes/notes/macos-sequoia-pop-up-fix-ffbb1b2447b059cd.yaml:11]:
```suggestion
    macOS Sequoia users will no longer receive a pop-up notification from the Agent asking for local network access.
```

**Review by drichards-87 — APPROVED** [review]
Left a suggestion from Docs and approved the PR.

**Comment by Pythyu** [comment]
After further testing it just seems that the pop-up notification can only happen the first time and we can't reset the permissions on a VM. So more as I expect it at first this PR actually does nothing 😞 , editing the PR...

**Pull request merged**